subsys/bluetooth/host/iso.c

```diff
@@ -1096,12 +1096,80 @@ void bt_iso_cleanup_acl(struct bt_conn *iso)
 	}
 }
 
-static void store_cis_info(const struct bt_hci_evt_le_cis_established *evt,
-			   struct bt_iso_info *info)
+static void store_cis_info(const struct bt_hci_evt_le_cis_established *evt, struct bt_conn *iso)
 {
+	struct bt_conn_iso *iso_conn = &iso->iso;
+	struct bt_iso_info *info = &iso_conn->info;
 	struct bt_iso_unicast_info *unicast_info = &info->unicast;
-	struct bt_iso_unicast_tx_info *central = &unicast_info->central;
 	struct bt_iso_unicast_tx_info *peripheral = &unicast_info->peripheral;
+	struct bt_iso_unicast_tx_info *central = &unicast_info->central;
+	const uint8_t c_phy = bt_get_phy(evt->c_phy);
+	const uint8_t p_phy = bt_get_phy(evt->p_phy);
+	struct bt_iso_chan_io_qos *tx;
+	struct bt_iso_chan_io_qos *rx;
+	struct bt_iso_chan *chan;
+
+	iso_conn = &iso->iso;
+	chan = iso_conn->chan;
+	rx = chan->qos->rx;
+	tx = chan->qos->tx;
+
+	LOG_DBG("iso_chan %p tx %p rx %p", chan, tx, rx);
+
+	if (iso->role == BT_HCI_ROLE_PERIPHERAL) {
+		/* As of BT Core 6.0, we can only get the SDU size if the controller
+		 * supports bt_hci_evt_le_cis_established_v2. Since this is not guaranteeds,
+		 * we fallback to using the PDU size as the SDU size.
+		 */
```

Comment:
I'm wondering if we should rather set it to a value which makes it clear that the SDU size is unknown

Reply:
Had the same thought. In the current version of Zephyr we do fallback to the PDU size, as that is the only value the CIS peripheral has. Setting it to an "UNKNOWN" value may prevent CIS peripherals from sending anything if they use the `sdu` field here, rather than the `pdu` field of the `info` struct

But I'm OK with either approach

Reply:
I would prefer setting it to to 0 when unknown. The app developer can use `can_send` and `can_recv` to detect if the SDU size is actually non-zero

Reply:
Since this would be unrelated to supporting this event, and require changes to tests and samples that use the qos.sdu value, I suggest that if we modify it, we do it in a separate PR, and keep existing functionality for this value